### ISSUE TITLE
ignore: Directory glob is too eager

### ISSUE THREAD

**Issue by Gilnaa** [Gilnaa/globwalk#10]
For the following dir structure:
```
mkdir target
touch {.,target}/{a,b,c}.rs
```

And the following settings:
```
GlobWalker::from_patterns(".", &["target/"])
```

The following Results are expected:
```
Ok(DirEntry("../testg/target"))
```

But extra results appear:
```
Ok(DirEntry("../testg/target"))
Ok(DirEntry("../testg/target/c.rs"))
Ok(DirEntry("../testg/target/b.rs"))
Ok(DirEntry("../testg/target/a.rs"))
```

This, and also #5 both stem from the use of `ignore` and the result negation.

**Comment by epage** [comment]
Starting to use #9 to explore fixing this.

**Comment by Gilnaa** [comment]
Closed by 30dd890f4a6ba5354c932dd71dc2cbbdda117091
